components/library/mozjs-60/Makefile

```diff
@@ -0,0 +1,126 @@
+COMPONENT_SRC=          mozjs-60.8.0
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.bz2
+COMPONENT_ARCHIVE_HASH= \
+	sha256:697331336c3d65b80ded9ca87b4a8ceb804e5342b476eaa133ac638102a9dc5d
+COMPONENT_ARCHIVE_URL=  http://ftp.gnome.org/pub/gnome/teams/releng/tarballs-needing-help/mozjs/$(COMPONENT_ARCHIVE)
+COMPONENT_FMRI= library/mozjs-60
+COMPONENT_SUMMARY= Javascript engine
+COMPONENT_CLASSIFICATION= Development/Other Languages
+COMPONENT_LICENSE= MPL v2.0
+COMPONENT_LICENSE_FILE= mozjs.license
+
+CXXFLAGS += -std=c++11
+
+# Only used during build process.
+PYTHON_VERSIONS= 2.7
+PYTHON= $(PYTHON.2.7.$(BITS))
+
+# Tests hook up need work
+TEST_TARGET = $(SKIP_TEST)
+
+include $(WS_MAKE_RULES)/common.mk
+
+xCOMPONENT_PRE_CONFIGURE_ACTION += ($(CLONEY) $(SOURCE_DIR) $(@D))
+
+COMPONENT_PRE_CONFIGURE_ACTION += ( \
+  cd $(COMPONENT_NAME)-$(COMPONENT_VERSION); \
+  cd js/src; \
+  autoconf-2.13 --localdir=. old-configure.in; \
+  touch configure; \
+  );
+
+# configure script is not at root of source tree.
+CONFIGURE_SCRIPT= $(SOURCE_DIR)/js/src/configure
+
+CONFIGURE_ENV += PYTHON="$(PYTHON)"
+CONFIGURE_ENV += LDSHARED="$(CC) -shared"
+CONFIGURE_ENV += NSPR_CONFIG=/usr/bin/amd64/nspr-config
+
+# detection of ARCH by config.guess depends on the compiler (e.g. gcc > 8)
+CONFIGURE_OPTIONS += --target=$(GNU_ARCH)
+CONFIGURE_OPTIONS += --disable-tests
+CONFIGURE_OPTIONS += --disable-static-js
+CONFIGURE_OPTIONS += --enable-optimize
+CONFIGURE_OPTIONS += --enable-readline
+CONFIGURE_OPTIONS += --enable-shared-js
+CONFIGURE_OPTIONS += --enable-threadsafe
+CONFIGURE_OPTIONS += --with-intl-api
+CONFIGURE_OPTIONS += --with-system-zlib
+CONFIGURE_OPTIONS += --with-x
+CONFIGURE_OPTIONS += --with-system-nspr
+
+COMPONENT_BUILD_ARGS += $(JOBS:%=-j%)
+
+COMPONENT_POST_INSTALL_ACTION  = \
+    for file in `find $(PROTO_DIR)$(CONFIGURE_LIBDIR.$(BITS)) -name "*.so"`; do \
+        /usr/bin/elfedit -e 'dyn:value -s  RUNPATH "$(GCC_LIBDIR):/usr/lib/mps/amd64"' $$file ; \
+        /usr/bin/elfedit -e 'dyn:value -s  RPATH   "$(GCC_LIBDIR):/usr/lib/mps/amd64"' $$file ; \
+    done ; \
+    for file in `find $(PROTO_DIR)$(USRBINDIR) -name "js60"`; do \
+        /usr/bin/elfedit -e 'dyn:value -s  RUNPATH "$(GCC_LIBDIR):/usr/lib/mps/amd64"' $$file ; \
+        /usr/bin/elfedit -e 'dyn:value -s  RPATH   "$(GCC_LIBDIR):/usr/lib/mps/amd64"' $$file ; \
+    done ;
+
+# js coredumps on sparc - see bug 24935318
+ifeq ($(MACH), sparc)
+LD_EXEC_OPTIONS.64 += -M $(WS_COMPONENTS)/desktop/polkit/mapfile
+COMPONENT_BUILD_ENV.64 += LD_EXEC_OPTIONS="$(LD_EXEC_OPTIONS.64)"
+COMPONENT_INSTALL_ENV.64 += LD_EXEC_OPTIONS="$(LD_EXEC_OPTIONS.64)"
+endif
+
+unexport SHELLOPTS
+COMPONENT_TEST_ARGS =   -k -i
+COMPONENT_TEST_TRANSFORMS += \
+        '-n ' \
+        '-e "/TOTAL:/p" ' \
+        '-e "/SKIP:/p" ' \
+        '-e "/PASS:/p" ' \
+        '-e "/FAIL:/p" ' \
+        '-e "/ERROR:/p" '
+
+# Build dependencies
+REQUIRED_PACKAGES += developer/build/autoconf-213
+
```

Comment:
Dependency on developer/gcc-8 is missing:

/usr/gcc/8/bin/gcc -fno-strict-aliasing -m32 -O3 -fPIC -DPIC -std=c99 -D_XOPEN_SOURCE=600 -D_REENTRANT -DNDEBUG -g -O3 -Wall -Wstrict-prototypes -KPIC -DPIC -DPSUTIL_POSIX=1 -DPSUTIL_VERSION=543 -DPSUTIL_SUNOS=1 -I/usr/include/python2.7 -c psutil/_psutil_common.c -o build/temp.solaris-2.11-i86pc.64bit-2.7/psutil/_psutil_common.o
unable to execute '/usr/gcc/8/bin/gcc': No such file or directory
error: command '/usr/gcc/8/bin/gcc' failed with exit status 1

Error processing command. Ignoring because optional. (optional:setup.py:third_party/python/psutil:build_ext:--inplace)
Error processing command. Ignoring because optional. (optional:packages.txt:comm/build/virtualenv_packages.txt)